src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -352,67 +352,60 @@ public BigInteger(ReadOnlySpan<byte> value, bool isUnsigned = false, bool isBigE
             }
             else
             {
-                int unalignedBytes = byteCount % 4;
-                int dwordCount = byteCount / 4 + (unalignedBytes == 0 ? 0 : 1);
-                uint[] val = new uint[dwordCount];
-                int byteCountMinus1 = byteCount - 1;
-
-                // Copy all dwords, except don't do the last one if it's not a full four bytes
-                int curDword, curByte;
+                int wholeDwordCount = Math.DivRem(byteCount, 4, out int unalignedBytes);
```

Comment:
Nit: it's better to use the term `uint` instead of `dword`.

Reply:
You can say `UInt32` if you want to explicitly see the bit size in the name.